src/LevelSets/compute_level_sets.jl

```diff
@@ -1,20 +1,25 @@
-@kernel inbounds = true function _init_level_set!(Ψ::Field, dem::Field, dem_grid::StructuredGrid, Ψ_grid::StructuredGrid, cutoff, R)
-    I = @index(Global, Cartesian)
-    x, y, z = coord(Ψ_grid, location(Ψ), I)
+"""
+    _init_level_set!(Ψ::Field, dem::Field, dem_grid::UniformGrid, Ψ_grid::UniformGrid, cutoff::AbstractFloat, R::AbstractMatrix)
+
+Initialize level sets.
+"""
+@kernel function init_level_set!(Ψ::Field, dem::Field, dem_grid::UniformGrid, Ψ_grid::UniformGrid, cutoff, R)
+    I = @index(Global, NTuple)
+    x, y, z = coord(Ψ_grid, location(Ψ), I...)
     P = R * Point3(x, y, z)
     ud, sgn = sd_dem(P, cutoff, dem, dem_grid)
-    Ψ[I] = ud * sgn
+    Ψ[I...] = ud * sgn
 end
 
+
 """
-    compute_level_set_from_dem!(arch, Ψ, dem, dem_grid, Ψ_grid, R[=LinearAlgebra.I])
+    compute_level_set_from_dem!(arch::Architecture, Ψ::Field, dem::Field, dem_grid::UniformGrid, Ψ_grid::UniformGrid, R=LinearAlgebra.I)
 
-Compute a level set from a given dem.
+Compute level sets from dem.
 """
-function compute_level_set_from_dem!(arch::Architecture, Ψ::Field, dem::Field, dem_grid::StructuredGrid, Ψ_grid::StructuredGrid,
-                                     R=LinearAlgebra.I)
-    cutoff = 4maximum(spacing(Ψ_grid))
-    kernel = _init_level_set!(backend(arch), 256, size(Ψ))
+function compute_level_set_from_dem!(backend, Ψ::Field, dem::Field, dem_grid::UniformGrid, Ψ_grid::UniformGrid, R=LinearAlgebra.I)
+    kernel = init_level_set!(backend, 256, size(Ψ))
+    cutoff = 4maximum(spacing(Ψ_grid, Center(), 1, 1, 1))
```

Comment:
This may fail if not in 3D

Reply:
You are right, @utkinis what would be the best way to fix?

Reply:
For UniformGrid, we now have a new method `spacing that should make it`